### ISSUE TITLE
gwt-elemental contains both class and java files in same JAR

### ISSUE THREAD

**Issue by Frettman** [vaadin/flow#8412]
The gwt-elemental artifact provided by Vaadin and a dependency of flow-server seems to have a packaging issue. The binary JAR **gwt-elemental-2.8.2.vaadin2.jar** as well as the sources JAR **gwt-elemental-2.8.2.vaadin2-sources.jar** both contain the class as well as the java files.
This leads to errors like this when running Javadoc (with Java 11):
`/project/src/main/java/my/package/MyClass.java:11: error: cannot access Json
 import elemental.json.Json;
                      ^
   bad source file: /project/.m2/repository/com/vaadin/external/gwt/gwt-elemental/2.8.2.vaadin2/gwt-elemental-2.8.2.vaadin2.jar(/elemental/json/Json.java)
     file does not contain class elemental.json.Json
     Please remove or make sure it appears in the correct subdirectory of the sourcepath.
`

**Comment by caalador** [comment]
Gwt requires the soures to be abbed to the jar file for compilation to JavaScript.
The vaadin jar version is built the same as the actual `gwt-elemental-2.8.2.jar` as you can see if you get the jar file for it from for instance https://mvnrepository.com/artifact/com.google.gwt/gwt-elemental/2.8.2

Most likely you just need to add the  property:
```xml
<properties>
    <additionalparam>-Xdoclint:none</additionalparam>
</properties>
```

**Comment by Frettman** [comment]
I already have my maven-javadoc-plugin configured with <doclint>none</doclint> and I still get the same error. I made sure that the doclint setting is actually used.
I assume the GWT-compatibility is only still needed for the Multiplatform Runtime? Maybe the source code artifact could be moved to the dependencies there?

**Comment by caalador** [comment]
The Flow ClientEngine is actually still compiled with GWT even if application developers do not need to do widgetset compilation.

**Comment by Frettman** [comment]
Interesting. I have noticed the generated Java code but didn't know it was created by GWT.

Anyway, I'm using Java 11, if that makes a difference. Meanwhile all attempts at making maven-javadoc-plugin ignore either gwt-elemental or the class using it didn't pan out so far. So I'm not generating a Javadoc artifact for now. I hope Vaadin's add-on directory doesn't require it :)

**Comment by caalador** [comment]
I'm quite certain Java11 has something to do with it (or Java9+), but we need to investigate the reason why the JavaDoc generation fails.

I checked the addon-starter pom and it seems to add the javadoc jar, but I would have expected it to not be needed. Else building the add-on in Java8 compatibility mode would be a possibility.

**Comment by Frettman** [comment]
I was compiling with
<maven.compiler.source>1.8</maven.compiler.source>
<maven.compiler.target>1.8</maven.compiler.target>
anyway. But it seems the Javadoc plugin only reacts to maven.compiler.release. After switching to that it works now.
Still, full Java 11 compatibility with regards to Javadoc would be nice at some point in the future, even though I mostly blame Javadoc for it's overly sensitive behavior.

**Comment by denis-anisimov** [comment]
I can't reproduce this issue with Java 11 and modified https://github.com/vaadin/addon-starter-flow.

So:
- first of all : having class files and java files in the same Jar is not a bug. It's necessary for GWT.
- It should be possible to generate javadocs even if you have Jars which has binaries and java sources.

The questions are:
- do you use your own project for add-on which has javadocs plugin or your project is based on https://github.com/vaadin/addon-starter-flow ?
- so technically the question is : is it an issue in `addon-starter-flow` or it's just something you are facing on your own?

I suppose that javadoc tries to compile the Java sources from the classpath which is used as a default value for `sourcepath` property value: https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#sourcepath
On the other hand: it should be the same regardless of Java version : 8 or 11.

But anyway : try to set the `sourcepath` config property in maven:
https://maven.apache.org/plugins/maven-javadoc-plugin/jar-mojo.html#sourcepath.

And finally: should this be an issue in the https://github.com/vaadin/addon-starter-flow repo in fact ?

**Comment by Frettman** [comment]
I started with the template project from https://vaadin.com/start/lts/component - which I suppose is the same as the linked Github repository - and then rearranged it to match my other non-Vaadin projects. But ultimately it's still a pretty standard Java Maven project.

I now tried to recreate the problem with a fresh addon starter by copying over my source, adding some dependencies, using the latest versions of the javadoc and compiler plug-ins and setting maven.compiler.release to 11. But Javadoc worked fine...
So I picked them both apart and camparing them bit by bit I found the critical difference:
I have added an `Automatic-Module-Name` to the Manifest in the `maven-jar-plugin`. Without it everything works as expected.

Either way, you are correct that it's not a problem with gwt-elemental. Judging from javac's command line flag [-Xprefer](https://docs.oracle.com/en/java/javase/11/tools/javac.html) having both source and binary of a class is explicitly supported.
So, from what I have gathered this has to be a Javadoc bug. The error message doesn't even make sense: it finds the source file and complains that it doesn't contain the expected class which it absolutely does.

Seeing as there isn't anything to fix on your end I'm closing this ticket. Hopefully, the next poor soul stumbling across this error finds this ticket :)

BTW, I tried sourcepath. Setting it to blank had no change. With a dummy value Javadoc wouldn't fail but also not create any API - duh. `${basedir}/src/main/java` triggered yet another error. So it didn't work as workaround for me.
